src/lib.rs

```diff
@@ -0,0 +1,230 @@
+        stutters: Option<&'a str>,
+        random: bool,
+    ) -> UwUify<'a> {
+        // I dislike this
+
+        let mut linkify = LinkFinder::new();
+        linkify.kinds(&[LinkKind::Email, LinkKind::Url]);
+        linkify.url_must_have_scheme(false);
+        let mut uwuify = UwUify {
+            text: text.unwrap_or_default(),
+            input: infile.unwrap_or_default(),
+            output: outfile.unwrap_or_default(),
+            linkify,
+            ..Default::default()
+        };
+
+        if random {
+            uwuify.floating_rng = Xoshiro256Plus::seed_from_u64(rand::rngs::OsRng.next_u64());
+            uwuify.int_rng = Xoshiro256PlusPlus::seed_from_u64(rand::rngs::OsRng.next_u64());
+        }
+
+        if let Some(words) = words {
+            uwuify.words = words.parse::<f64>().unwrap();
+        }
+        if let Some(faces) = faces {
+            uwuify.faces = faces.parse::<f64>().unwrap();
+        }
+        if let Some(actions) = actions {
+            uwuify.actions = actions.parse::<f64>().unwrap();
+        }
+        if let Some(stutters) = stutters {
+            uwuify.stutters = stutters.parse::<f64>().unwrap();
+        }
+        uwuify
+    }
+
+    pub fn uwuify(&mut self) -> Result<(), Error> {
+        // Handle Text
+        if !self.text.is_empty() {
+            // Handle Text Output
+            if !self.output.is_empty() {
+                if Path::new(&self.output).exists() {
+                    return Err(Error::new(
+                        std::io::ErrorKind::AlreadyExists,
+                        format!("File '{}' already exists, aborting operation", &self.output),
+                    ));
+                }
+
+                let mut uwu_out_file = BufWriter::new(File::create(&self.output)?);
+                let uwu_progress_bar = progress_bar!();
+                self.uwuify_sentence(self.text, &mut uwu_out_file)?;
+
+                uwu_progress_bar.finish_with_message("UwU'ifying Complete!");
+                Ok(())
+            } else {
+                #[cfg(not(test))]
+                let stdout = std::io::stdout();
+                #[cfg(not(test))]
+                let mut out = BufWriter::new(stdout.lock());
+                #[cfg(test)]
+                let mut out = std::io::sink();
+                self.uwuify_sentence(self.text, &mut out)?;
+                #[cfg(not(test))]
+                out.write_all(b"\n")?;
+                Ok(())
+            }
+        } else {
+            // Handle File I/O
+            if Path::new(&self.output).exists() {
+                return Err(Error::new(
+                    std::io::ErrorKind::AlreadyExists,
+                    format!("File '{}' already exists, aborting operation", &self.output),
+                ));
+            }
+
+            let uwu_progress_bar = progress_bar!();
+            self.uwuify_sentence(
+                unsafe {
+                    std::str::from_utf8_unchecked(
```

Comment:
Why not validate utf-8 string beforehand?

Reply:
Ehh... It just doesn't matter too much if the file is valid UTF-8 or not. It's still UwU all the same

Reply:
Huh right, in my initial implementation. I was getting errors related to the byte size of each char being unknown. Does this also guarantee that when calling e.g `word[index]` we'll get the correct size of bytes for a char?

Reply:
Most likely. But like, if you're passing binary data into this program, then what do you expect?

Reply:
Yeah I guess, in my head, I just never want the program to ever panic or throw errors with stacktraces. It should always bubble up to the user in a nicely formatted way which clap allows us to do.

Reply:
I fixed up the error handling so that any and *all* errors are formatted nicely

Reply:
Ahh yeah I just saw that, I'm guessing that's this change here? -> https://github.com/sgoudham/uwuifyy/pull/2/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR97

Reply:
yes